src/definitions.jl

```diff
@@ -421,6 +421,10 @@ Broadcast.broadcasted(::typeof(*), x::Number, f::Frequencies) = Broadcast.broadc
 Broadcast.broadcasted(::typeof(/), f::Frequencies, x::Number) = Frequencies(f.n_nonnegative, f.n, f.multiplier / x)
 Broadcast.broadcasted(::typeof(\), x::Number, f::Frequencies) = Broadcast.broadcasted(/, f, x)
 
+Base.maximum(f::Frequencies) = ifelse(f.multiplier >= 0, (f.n_nonnegative - 1) * f.multiplier, (f.n_nonnegative - f.n) * f.multiplier)
```

Comment:
```suggestion
Base.maximum(f::Frequencies) = (f.n_nonnegative - ifelse(f.multiplier >= 0, 1, f.n)) * f.multiplier
```
This might improve things further, since no actual computation happens within the conditional, and `ifelse` permits evaluating all arguments.

Reply:
Thanks for the suggestion! I have implemented this in the latest commit